src/ppx_tools/dumpast.ml

```diff
@@ -0,0 +1,389 @@
+    method! object_field_desc x x' =
+      try super#object_field_desc x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#object_field_desc x
+
+    method! row_field_desc x x' =
+      try super#row_field_desc x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#row_field_desc x
+
+    method! core_type_desc x x' =
+      try super#core_type_desc x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#core_type_desc x
+
+    method! payload x x' =
+      try super#payload x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#payload x
+
+    method! constant x x' =
+      try super#constant x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#constant x
+
+    method! injectivity x x' =
+      try super#injectivity x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#injectivity x
+
+    method! arg_label x x' =
+      try super#arg_label x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#arg_label x
+
+    method! variance x x' =
+      try super#variance x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#variance x
+
+    method! closed_flag x x' =
+      try super#closed_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#closed_flag x
+
+    method! override_flag x x' =
+      try super#override_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#override_flag x
+
+    method! virtual_flag x x' =
+      try super#virtual_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#virtual_flag x
+
+    method! mutable_flag x x' =
+      try super#mutable_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#mutable_flag x
+
+    method! private_flag x x' =
+      try super#private_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#private_flag x
+
+    method! direction_flag x x' =
+      try super#direction_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#direction_flag x
+
+    method! rec_flag x x' =
+      try super#rec_flag x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#rec_flag x
+
+    method! longident x x' =
+      try super#longident x x' with
+      | Reparse_error m ->
+        warn_ast_diff m;
+        parse_ast#longident x
+  end
+
+let transform source kind =
+  let open Pp_path in
+  try
+    match kind with
+    | Structure ->
+      let v = Parse.implementation (Lexing.from_string source) in
+      parse_ast#structure v
+    | Signature ->
+      let v = Parse.interface (Lexing.from_string source) in
+      parse_ast#signature v
+    | Unknown -> Jsonoo.Encode.string "Unknown file extension"
+  with
+  | Syntaxerr.Error _ -> Jsonoo.Encode.null
```

Comment:
Perhaps we need a wrapper around the parse functions that returns a result type? So that we don't forget checking this error. But to me, this looks like another case where we should just abort and let the error bubble up. The uninformative null is suspicious again.

Reply:
Wrapped in result. 70503e6382b3377d2040697f280b4a9028039def
I think it would be fine to also indicate syntax error inside the AST explorer (at the moment nothing happends).